gprofiler/application_separators.py

```diff
@@ -0,0 +1,232 @@
+import configparser
+import os.path
+import re
+from abc import ABCMeta, abstractmethod
+from typing import List, Optional
+
+from granulate_utils.linux.ns import resolve_host_path
+from psutil import NoSuchProcess, Process
+
+from gprofiler.log import get_logger_adapter
+
+logger = get_logger_adapter(__name__)
+
+
+_PYTHON_BIN_RE = re.compile(r"^python([23](\.\d{1,2})?)?$")
+
+
+def _is_python_bin(bin_name: str):
+    return _PYTHON_BIN_RE.match(os.path.basename(bin_name)) is not None
+
+
+class ApplicationSeparator(metaclass=ABCMeta):
+    @abstractmethod
+    def is_supported(self, process: Process) -> bool:
```

Comment:
Maybe merge it into `get_application_name`? In some of the impls, e.g `PythonModuleApplicationSeparator`, you are making assumptions in `get_application_name` based on what was checked in `is_supported`. Code is better understood if you make the checks then immediately use the new "assumptions".